keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/SecurityGroupHandler.kt

```diff
@@ -167,7 +141,7 @@ open class SecurityGroupHandler(
       )
     }
 
-    val base = securityGroups.values.first()
+    val base = securityGroups.values.minByOrNull { it.inboundRules.size }!!
```

Comment:
This makes it so that the group with the fewest inbound rules is used as the default at the top-level of the spec, with overrides having the extra rules, since there's no way for an override to remove a rule (we merge them together).

Reply:
Is there any guarantee the smallest set is common to all though?

Reply:
No, but neither is any other random pick, and definitely not the largest. 🙂 
So I think you just identified a bug in our logic for calculating overrides... We can look into that in a separate PR.